scripts/apps/monitoring/controllers/AggregateCtrl.js

```diff
@@ -99,10 +102,42 @@ export function AggregateCtrl($scope, api, desks, workspaces, preferencesService
                                 return {'type': 'workspace', 'groups': []};
                             });
                     } else if (activeWorkspace.type === 'desk') {
-                        var desk = self.deskLookup[activeWorkspace.id];
-                        if (desk && desk.monitoring_settings) {
-                            return {'type': 'desk', 'groups': desk.monitoring_settings};
-                        }
+                        return preferencesService.get(PREFERENCES_KEY)
+                            .then(function(preference) {
+                                var desk = self.deskLookup[activeWorkspace.id];
+
+                                // Read available groups from user preferences first
+                                if (preference && preference[activeWorkspace.id] && preference[activeWorkspace.id].groups) {
```

Comment:
Please simplify this condition and try to use less indentation if possible. It would be very helpful. It's a bit difficult to follow.